src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -192,6 +195,12 @@ private unsafe int IndexOfOrdinalIgnoreCaseHelper(ReadOnlySpan<char> source, Rea
                 Next: ;
                 }
 
+                // Before we return -1, check if the remaining source contains any special or non-Ascii characters.
+                if (remainingSource.ContainsAnyExcept(s_nonSpecialAsciiChars))
```

Comment:
Do we really need to go through the whole `remainingSource` again? It looks expensive.

Reply:
`remainingSource` is the remaining part of the source which has not been processed before. So, we are not going through it again.